### PR TITLE
refactor: adopt parcel toolchain

### PR DESCRIPTION
**Pull request by LeoDog896**
This uses the [Parcel](https://parceljs.org/) toolchain, with the following refactrs:
- Offsets copying the book to parcel
- Moves the favicon images to one, resizable image
- Uses Split.js's [npm module](https://www.npmjs.com/package/split.js) instead of browser script

Current known issues with a full move to parcel (thus why partial codemirror src && some image assets are still there):
- Codemirror v5's npm module just acts *weird*, and doesn't work even if you import it without parcel's bundling feature. It's most likely necessary to make another PR that refactors to codemirror v6.
- when using [query resizing](https://parceljs.org/recipes/image/#resizing-and-converting-images) (`?width=X&height=Y`), mstile causes duplicate assets in parcel (https://github.com/parcel-bundler/parcel/issues/8821#issuecomment-1423195983) meaning that not all of the mstile images could be moved (excluding the 310x150 long image)

Note: The CI fails in this PR as I've added an extra action to install PNPM.

I've mostly avoided any script refactors as requested (#21), and will wait to the next PR to refactor scripts.

### CODE REVIEW

**Comment by tomtau** [comment]
`"pnpm": executable file not found in $PATH`

do GH actions need to be updated as well?

**Comment by LeoDog896** [comment]
> `"pnpm": executable file not found in $PATH`
> 
> the setup action does not set the PATH?

Pull Requests that update CI don't actually update the repository CI. Take for example #23 -- no CI was attached to any of the commits.

File: static/index.html [hunk 1]
```diff
@@ -398,6 +396,7 @@ <h2>Measurements</h2>
         if (window.lint) {
           var doc = CodeMirror.Doc(text);
           var errors = window.lint(text);
+          console.log(errors)
```
Comment (tomtau) [inline static/index.html:399]: Needed for debugging on the deployed version or should that be removed?

Comment (LeoDog896) [reply static/index.html:399]: It should be removed.

**Comment by tomtau** [comment]
@LeoDog896 got this error: https://github.com/pest-parser/site/actions/runs/4205350448/jobs/7297306521#step:7:52 so will revert the commit for now:
```
[INFO]: :-) Your wasm pkg is ready to publish at /home/runner/work/site/site/pkg.
task: [default] pnpm run build

> @ build /home/runner/work/site/site
> parcel build static/index.html

@parcel/core: Cannot find extended parcel config

/home/runner/work/site/site/.parcelrc:2:14
  1 | {
> 2 |   "extends": "@parcel/config-default",
>   |              ^^^^^^^^^^^^^^^^^^^^^^^^ Cannot find module "@parcel/config-default"
  3 |   "transformers": {
  4 |     "raw:*": ["@parcel/transformer-raw"],

 ERROR  Command failed with exit code 1.
task: Failed to run task "default": exit status 1
```

**Comment by LeoDog896** [comment]
Alright, I'll reopen the PR once I can reproduce this.

**Pull request merged**